rfcs/0000-policy_target_scope.md

```diff
@@ -0,0 +1,267 @@
+
+The resolving of _Policies_ to enforce occurs at the `HTTPRoute` or within individual `HTTPRouteRule`
+within a route. The merging strategy will resolve the individual `HTTPRouteRule` all the policies boil
+down to, stack them according to the `default` and `override` and appy the most specific one according to
+the Gateway API Policy Attachment hierarchy. Targeting a complete `HTTPRoute` is the equivalent of
+targeting each indidividual `HTTPRouteRule`s that composes the route.
+
+## What actual use-cases does this address?
+
+ - An admin restricts all traffic using a `default` at the gateway to force application developers to
+   "think" about how to authorize access to their services and have them come up with sensible `default`
+   on their `HTTPRoute`s
+ - An application developer protects its service by some rate limit, while an cluster admin could come
+   and use an `override` to mitigate a DDoS.
+
+# Reference-level explanation
+[reference-level-explanation]: #reference-level-explanation
+
+So there are a couple of things to consider here:
+
+- status reporting on "not yet" existing `HTTPRouteRule`, e.g. a Policy declares a set of `matches` that
+  don't (yet?) exist. The policy must _not_ be enforced... yet. But could eventually. This needs proper
+  reporting. 
+
+# Drawbacks
+[drawbacks]: #drawbacks
+
+- The use cases this approach doesn't address… 
+- Not very DRY... the need to keep `HTTPRouteRule`s in sync between the Policy and the `HTTPRoute` is
+  suboptimal.
+
+# Rationale and alternatives
+[rationale-and-alternatives]: #rationale-and-alternatives
+
+- [`HTTPRouteFilter`](https://gateway-api.sigs.k8s.io/references/spec/#gateway.networking.k8s.io/v1beta1.HTTPRouteFilter)
+  would be an alternative and provide finer grained control over how a policy applies, i.e. at the
+  `HTTPRouteRule` level, or even on a per `HTTPBackendRef` within an `HTTPRouteRule`. But today there is
+  no way (that I know of) for us to add a `HTTPRouteFilterType` to the list of supported
+  `HTTPRouteFilter` by a Gateway API implementor. 
+
+# Prior art
+[prior-art]: #prior-art
+
+I don't know of prior art per se, _but_ the envoy gateway project, if only for rate limiting, seems to
+[avoid policy attachment](https://github.com/envoyproxy/gateway/issues/670#issuecomment-1299114982), but
+they are an implementor and as such can extend the
+[`HTTPRouteFilter`](https://gateway-api.sigs.k8s.io/references/spec/#gateway.networking.k8s.io/v1beta1.HTTPRouteFilter)
+types. This proposed approach if actually trying to enable achieving the same control `HTTPRouteFilter`
+gives, while using _Policy Attachement_ and targeting a specific `HTTPRouteRule` within a `HTTPRoute`.
+
+# Unresolved questions
+[unresolved-questions]: #unresolved-questions
+
+- What use cases are we making "overly" complex, because of the duplication of `HTTPRoute`'s
+  `HTTPRouteMatch` ?
+- Is it fine for the `matches` section to be at the same level as `default`, `override` and `targetRef`?
```

Comment:
I think the fact it is an optional complexity is the perfect balance at least until we hear otherwise via new use cases. You have the "target all of a resource by default" and given a way to cover more advanced use cases and targeting. I also like the fact that the default section stays focused on the actual domain (RL) and leaves the targeting as a separate piece